### PR TITLE
feat: add disabled prop

### PR DESCRIPTION
**Pull request by cwjTerrace**
```
function Text(props: TextProps) {
  const timeout = useRef<number>();
  const [success, setSuccess] = useState(false);
  const { children, copyable, ...rest } = props;

  const cleanCopyId = () => {
    clearTimeout(timeout.current!);
  };

  React.useEffect(() => cleanCopyId, []);

  if (copyable) {
    const { iconSize = 'small', ...copyPropsRest } = copyable as CopyProps;
    const copyProps = {
      ...copyPropsRest,
      text: copyPropsRest.text || String(children) || '',
    };

    const CopyIcon = (
      <Icon type={!success ? 'copy' : 'check'} size={iconSize} className={genClass('copy-icon')} />
    );

    return (
      <WrapperText {...rest}>
        {children}
        {success ? (
          CopyIcon
        ) : (
          <CopyToClipboard
            {...copyProps}
            onCopy={(text: string, result: boolean) => {
              if (result) {
                setSuccess(result);
                cleanCopyId();
                timeout.current = window.setTimeout(() => setSuccess(false), 3000);
              }
              copyProps.onCopy?.(text, result);
            }}
          >
            {CopyIcon}
          </CopyToClipboard>
        )}
      </WrapperText>
    );
  }

  return <WrapperText {...rest}>{children}</WrapperText>;
}
```

```
function Text(props: TextProps) {
  const timeout = useRef<number>();
  const [success, setSuccess] = useState(false);
  const { children, copyable, ...rest } = props;

  const cleanCopyId = () => {
    clearTimeout(timeout.current!);
  };

  React.useEffect(() => cleanCopyId, []);

  if (copyable) {
    const { iconSize = 'small', ...copyPropsRest } = copyable as CopyProps;
    const copyProps = {
      ...copyPropsRest,
      text: copyPropsRest.text || String(children) || '',
    };

    const CopyIcon = (
      <Icon type={!success ? 'copy' : 'check'} size={iconSize} className={genClass('copy-icon')} />
    );

    return (
      <WrapperText {...rest}>
        {children}
        <CopyToClipboard
          {...copyProps}
          disabled={success}
          onCopy={(text: string, result: boolean) => {
            if (result) {
              setSuccess(result);
              cleanCopyId();
              timeout.current = window.setTimeout(() => setSuccess(false), 3000);
            }
            copyProps.onCopy?.(text, result);
          }}
        >
          {CopyIcon}
        </CopyToClipboard>
      </WrapperText>
    );
  }

  return <WrapperText {...rest}>{children}</WrapperText>;
}
```

### CODE REVIEW

File: src/Component.js [hunk 1]
```diff
@@ -51,10 +53,11 @@ export class CopyToClipboard extends React.PureComponent {
       onCopy: _onCopy,
       options: _options,
       children,
+      disabled,
       ...props
     } = this.props;
     const elem = React.Children.only(children);
 
-    return React.cloneElement(elem, {...props, onClick: this.onClick});
+    return React.cloneElement(elem, {...props, onClick: !disabled ? this.onClick : undefined});
```
Comment (nkbt) [inline src/Component.js:61]: All the props are passed through, including disabled

**Review by nkbt — CHANGES_REQUESTED** [review]
Absolutely unnecessary

File: src/Component.js (hunk 1 above)
Comment (cwjTerrace) [reply src/Component.js:61]: But disabled can't prevent onclick

Comment (nkbt) [reply src/Component.js:61]: It shouldn't. Up to the app to implement it, not the lib.

**Pull request closed without merging**